src/binding.jl

```diff
@@ -66,6 +66,21 @@ function bind_type(location, T, input, binder)
 
     bound_type
 end
+function try_bind_type(location, T, input, binder)
+    # bind type at macro expansion time.  It will be verified at runtime.
```

Comment:
This comment is a wrong, since code to verify the type is injected elsewhere, after binding the type again elsewhere in the code. Perhaps just

```julia
# bind type at macro expansion time in the caller's module.
```